crates/wasmtime/src/linker.rs

```diff
@@ -270,6 +271,117 @@ impl Linker {
         Ok(self)
     }
 
+    /// Define automatic instantiations of a [`Module`] in this linker.
+    ///
+    /// For a Reactor module, this instantiates the module and adds all its
+    /// exports to the `Linker` as the `instance` method.
```

Comment:
The terms "reactor" and "command" I think are pretty new to the documentation here. Would it be possible to link to somewhere else that has more details? This could also clarify that non-WASI-specific modules are handled here too, and they're treated similarly as to reactors.

Reply:
I think in general though these two paragraphs about what happens for reactor/commands could be expanded to describe in a bit more detail what's happening under the hood.

Reply:
I've now expanded this comment accordingly.